torch/export/export_schema.py

```diff
@@ -0,0 +1,302 @@
+    List[SymIntArgument],   # Symint[] can be an argement, there are symint[] in native_function.yaml
+
+    Scalar,         # !!! Scalar is already an union type: Union[int, float, bool], check if serialization library can handle this
+    # List[Scalar], # !!! for Scalar[], used in native_function.yaml, but not used in canonical aten ops yet... Consider if we need this
+    bool,
+    List[bool],     # for bool[]
+                    # !!! There are use of bool[3] in canonical aten ops, consider if we can simplify this
+    int,
+    List[int],      # for int[]
+    float,
+    List[float],    # for float[]
+    str,
+    # List[str],    # !!! There is no str[] in native_function.yaml. Consider if this is needed for expressiveness
+
+    # Graph,            # !!! Consider how to handle condition op, which need to pass in a graph for the branch
+    # List[Graph],      # !!! What about list of graphs? Do we need this?
+
+    # !!! Following types doesn't have a list version in native_function.yaml
+    ScalarType,
+    MemoryFormat,
+    Layout,
+    Device,
+]
+
+# !!! How to model optional fields? Is it an operator schema annotation, or an argument type?
+#     Tensor?
+#     Scalar?
+#     ScalarType?
+#     bool?
+#     int?
+#     int[]?
+#     float[]?
+#     SymInt[]?
+#     MemoryFormat?
+#     Layout?
+#     Device?
+
+
+################################################################################
+# Following section is the defining the schema of serializing a concrete tensor
+
+# TensorMeta is a decription of a tensor, without the actual data (,effectively maps to FakeTensor)
+# TensorMeta has multliple uses
+#   1. Represent the property of a concrete tensor backed by a storage
+#     - This is used in the serialization of a concrete tensor, e.g. model weight
+#     - In this case, sizes and strides must be concrete ints, and cannot be symbolic
+#     - stride and storage_offset have to used to correctly reconstruct the tensor from the storage
+#   2. Represent the property of a virtual tensor (see IValue below)
+#     - In this case, sizes and strides can be either concrete ints or symbolic ints.
+#     - device/strides/storage_offset/layout/memory_format are tied to pytorch's implementation.
+#       These are faithful capture of tensor's detail in pytorch's executions during tracing
+#       However, it's up to downstream system on how to utilized these fields
+#       In another word, these feilds are suggestive, rather than mandatory.
+
+@dataclass
+class TensorMeta:
+    dtype: ScalarType
+    sizes: List[SymInt]
+
+    # needed for training
+    requires_grad: bool
+
+    # !!! see description above, there are subtle difference on how these fields should be interpreted
+    device: Device
+    strides: List[SymInt]
+    storage_offset: int
+    layout: Layout
+
+
+@dataclass
+class Buffer:
+    # !!! TODO: need to define endianness for the buffer
+    buffer: bytes
+
+
+@dataclass
+class ExternalBuffer:
```

Comment:
Add comment: This is intended to be used together with torch.storage.from_buffer

See https://pytorch.org/docs/stable/storage.html#torch.TypedStorage.from_buffer